wgengine/router/router_linux.go

```diff
@@ -430,6 +433,30 @@ func (r *linuxRouter) Set(cfg *Config) error {
 	return multierr.New(errs...)
 }
 
+var warnStatefulFilteringWithDocker = health.NewWarnable()
+
+func (r *linuxRouter) updateStatefulFilteringWithDockerWarning() {
+	// If stateful filtering is disabled, clear the warning.
+	if !r.statefulFiltering {
+		r.health.SetWarnable(warnStatefulFilteringWithDocker, nil)
+		return
+	}
+
+	// Otherwise, check for the presence of a Docker interface and warn if
+	// it's found on the system.
+	//
+	// TODO(andrew-d): do a better job at detecting Docker, e.g. by looking
+	// for it in the $PATH or by checking for the presence of the Docker
+	// socket/daemon/etc.
+	ifstate := r.netMon.InterfaceState()
+	if _, found := ifstate.Interface["docker0"]; found {
+		r.health.SetWarnable(warnStatefulFilteringWithDocker, fmt.Errorf(""+
+			"Stateful filtering is enabled and Docker was detected; this may prevent Docker containers "+
+			"on this host from connecting to Tailscale nodes.",
```

Comment:
If you want, you can add a shortlink. From the client we only link to `https://tailscale.com/s/FOO` and then you can go change FOO over time in tailscale-www redirects.

Reply:
(We'll later be making all Warnables have a URL for more info)

Reply:
Added a shortlink; will follow up with a PR to add it to the website later today.